### PR TITLE
Docs: More contribution details

### LINKED ISSUE #347: Provide more contribution related details
- [x] Add `ISSUE_TEMPLATE.md`.
- [x] Add `PULL_REQUEST_TEMPLATE.md`.
- [x] Add [`CODEOWNERS` file](https://github.com/blog/2392-introducing-code-owners) (see: #382).
- [ ] Update `CONTRIBUTING.md` with more information.
- [ ] Add [`SUPPORT` file](https://github.com/blog/2400-support-file-support).

### PR DESCRIPTION
**Pull request by molant**
Fix #347

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.

## Short description of the change(s)

Add a few templates for the most common questions/requests so things are better organized. Maybe we want to consolidate more and add webhint.io issues in here and move now that GitHub allows to do that? Or maybe a template that tells the user to open it in that repo?

We can probably close the rfcs repo after moving the items here. Don't know if we'll need a few more labels.

This is how it looks like. You can check it by going to https://github.com/molant/hint/issues/new

image

### CODE REVIEW

**Comment by brendyna** [comment]
I'm in favor of having as few repos as possible (unless I'm missing a reason to have more). It took me a bit to get the lay of the land and understand where/how to give feedback. But, even if we don't consolidate, I like the idea of a link on this page for providing other types of feedback and then using one page (e.g. this one) whenever we link to "provide webhint feedback" so it's a single source of truth.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -3,6 +3,18 @@
 Please take a moment to review this document in order to make the
 contribution process easy and effective for everyone involved.
 
+## Contributing code
+
+Before doing any major changes, there are a few things you should do:
+
+1. Open an issue to discuss and potentially help if the team agrees
+   the changes will be for the best.
+1. Familiarize yourself with the architecture and concepts by reading
```
Comment (brendyna) [inline .github/CONTRIBUTING.md]: 2

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -3,6 +3,18 @@
 Please take a moment to review this document in order to make the
 contribution process easy and effective for everyone involved.
 
+## Contributing code
+
+Before doing any major changes, there are a few things you should do:
```
Comment (brendyna) [inline .github/CONTRIBUTING.md]: A couple (not a "few" since there are only two)

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (brendyna) [reply .github/CONTRIBUTING.md]: Also, "reading _the_ contributor guide"

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -3,6 +3,18 @@
...
+
+1. Open an issue to discuss and potentially help if the team agrees
+   the changes will be for the best.
+1. Familiarize yourself with the architecture and concepts by reading
+   [contributor guide][contributor guide] and the projects
+   [pull request process][pr process].
+
+We thank you for following this tips!
```
Comment (brendyna) [inline .github/CONTRIBUTING.md]: "these" tips

File: .github/ISSUE_TEMPLATE/1-bug-report.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+---
+name: "\U0001F41EBug report"
+about: Report a bug in webhint
+---
+<!--
+
+Hi there 👋
+
+We are sorry you've found an issue in our project. Before
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/1-bug-report.md]: Would be fun to frame this more positively: "Thanks for taking the time to make webhint better!"

File: .github/ISSUE_TEMPLATE/1-bug-report.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ones (even closed!) as they might contain information about
+workarounds, resolution, or progress updates.
+
+-->
+
+# 🐛 Bug report
+
+## Description
+
+<!-- ✍️ What were you doing? -->
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/1-bug-report.md]: In addition to "What were you doing?" how about something like: "What were you doing? What do we need to do to experience the issue you saw?"

File: .github/ISSUE_TEMPLATE/1-bug-report.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+## Environment
+
+* __`hint` version:__ <!-- ✍️ Run `hint -v` -->
+* __`Node.js` version:__ <!-- ✍️ Run `node -v` -->
+* __`npm` version:__ <!-- ✍️ Run `npm -v ` -->
+* __URL for which webhint failed:__
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/1-bug-report.md]: Is this always applicable?

File: .github/ISSUE_TEMPLATE/1-bug-report.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+## `webhint` configuration
+
+<details>
+<summary>webhint’s configuration</summary>
+
+```js
+<!-- Paste the content of your`.hintrc` here -->
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/1-bug-report.md]: Space after "your"

File: .github/ISSUE_TEMPLATE/1-bug-report.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+```
+
+</details>
+
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/1-bug-report.md]: Is this new-line extra?

File: .github/ISSUE_TEMPLATE/2-feature-request.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+---
+name: "\U0001F680Feature request"
+about: Suggest a feature for webhint
+
+---
+<!--
+
+Hi there 👋
+
+Before opening a new issue please make sure to search in the existing
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/2-feature-request.md]: How about an opener something like "We're excited for your ideas about how to make webhint even better! Before opening a new issue..."

File: .github/ISSUE_TEMPLATE/2-feature-request.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+# 🚀 Feature request
+
+## Description
+
+<!-- ✍️ --> What do you want to be added?
+
+<!-- 💪 --> Are you willing to work on this?
+
+## What scenarios will this solve?
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/2-feature-request.md]: I'd love if we put this first. Thinking about and articulating the problem more clearly often results in a better, more articulated solution.

File: .github/ISSUE_TEMPLATE/3-docs-bug.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
+---
+name: "📚 Docs issue"
+about: Report an issue in webhint's documentation
+
+---
+<!--
+
+Hi there 👋
+
+Before opening a new issue please make sure to search in the existing
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/3-docs-bug.md]: A positive, thankful opening here would be nice (similar to other suggestions above)

File: .github/ISSUE_TEMPLATE/4-new-hint-request.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
+---
+name: "\U0001F4A1Hint request"
+about: Suggest a new hint for webhint
+
+---
+<!--
+
+Hi there 👋
+
+Before opening a new issue please make sure to search in the existing
```
Comment (brendyna) [inline .github/ISSUE_TEMPLATE/4-new-hint-request.md]: Another positive, thankful opening would be nice (similar to those above). _Especially_ for new hints we should be thankful as we'll benefit a lot from crowdsourcing the web for these (as we already have).

**Comment by molant** [comment]
@brendyna feedback in. Feel free to use the suggestions button to propose edits directly:
image

File: .github/CONTRIBUTING.md [hunk 13]
```diff
@@ -3,6 +3,18 @@
 Please take a moment to review this document in order to make the
 contribution process easy and effective for everyone involved.
 
+## Contributing code
```
Comment (antross) [inline .github/CONTRIBUTING.md]: There's also the existing "Contributing Code" section further down. We should probably merge the two or give them different titles.

File: .github/ISSUE_TEMPLATE/1-bug-report.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+## Description
+
+<!-- ✍️ What were you doing? How do we repro your issue? -->
+
+## Environment
+
+* __`hint` version:__ <!-- ✍️ Run `hint -v` -->
+* __`Node.js` version:__ <!-- ✍️ Run `node -v` -->
+* __`OS`:__ <!-- ✍️ -->
+* __URL for which webhint failed:__ <!-- ✍️ -->
```
Comment (antross) [inline .github/ISSUE_TEMPLATE/1-bug-report.md]: Most of these pieces of environment data will only apply for the CLI (as will other details such as running `--debug` further down). When a user is reporting a bug in the online scanner, VS Code extension, or the browser extension we'll probably want to ask for slightly different information. 

I could easily see having separate "report a bug" templates for each of these targets (though we don't necessarily need to add all of them right now). Thoughts?

File: .github/ISSUE_TEMPLATE/2-feature-request.md (hunk 10 above)
Comment (antross) [reply .github/ISSUE_TEMPLATE/2-feature-request.md]: Asking for scenarios even after the description should help encourage this thought process for whoever files the request. I think leading with the description will help the rest of us grok the request more quickly after it gets filed.

File: .github/ISSUE_TEMPLATE/1-bug-report.md (hunk 14 above)
Comment (molant) [reply .github/ISSUE_TEMPLATE/1-bug-report.md]: Yes, we can probably have separate templates for those and add them once they are more broadly used.

File: .github/CONTRIBUTING.md (hunk 13 above)
Comment (molant) [reply .github/CONTRIBUTING.md]: I've merged both sections.

**Comment by molant** [comment]
> I'm really looking forward to turning this on. Now if only GitHub would let us [automatically add labels](https://github.community/t5/How-to-use-Git-and-GitHub/Apply-labels-automatically-at-issue-creation/td-p/10719) as well (maybe soon)...

🤞🤞🤞

@antross this should be ready for another review. Feel free to squash whenever you are ok with this.

**Pull request merged**